src/runtime/primitives/system_call.zig

```diff
@@ -88,7 +89,16 @@ pub fn Read_BytesInto_AtOffset_From_IfFail(context: *PrimitiveContext) !Executio
     const arguments = context.getArguments("_Read:BytesInto:AtOffset:From:IfFail:");
     const bytes_to_read = try arguments.getInteger(0, .Unsigned);
     const byte_array = try arguments.getObject(1, .ByteArray);
-    const offset: usize = @intCast(try arguments.getInteger(2, .Unsigned));
+    const offset_int = try arguments.getInteger(2, .Unsigned);
+    if (exceedsBoundsOf(offset_int, usize)) {
+        return ExecutionResult.completion(try Completion.initRuntimeError(
+            context.vm,
+            context.source_range,
+            "Offset argument is larger than the maximum value of usize",
+            .{},
+        ));
```

Comment:
Can we make this error a bit clearer? Users of the VM may not know what `usize` means.

```suggestion
        return ExecutionResult.completion(try Completion.initRuntimeError(
            context.vm,
            context.source_range,
            "Offset argument is larger than the maximum value allowed by your platform ({})",
            .{std.math.maxInt(usize)},
        ));
```